### ISSUE TITLE
Maria-Rosa asked for an aperMag2 column to be added to the vmcSource in the release as the source density is very high. I think we can rerun the source table (it should be the same sources and replace it).

### ISSUE THREAD

**Issue by wfastrononomer** [wfau/ScienceArchives#1012]
We already use smaller apertures for VVV and GPS, so it should be straightforward.

**Comment by wfastrononomer** [comment]
They show plots using aperMag2 in the release paper they are preparing.

**Comment by wfastrononomer** [comment]
VVV/VVVX use aperMag [1,3,4]. Code is in VSA_autoTemplateSchema.
Easy enough to add an extra modification for vmcDeep.

**Comment by wfastrononomer** [comment]
Modified schema - but getting issue when running mkmerge:

.# 2026-07-09 13:34:49.48: <INFO> Instantiated method 'ColourIndex' for targets: jmksExt
.# 2026-07-09 13:34:49.48: # --------------------------------------------------------
.# 2026-07-09 13:34:49.48: # A wfcamsrc exception was caught!
.# 2026-07-09 13:34:49.48: # DataMethodException: ColourIndex method : cannot source; -1;-1
.# 2026-07-09 13:34:49.48: # --------------------------------------------------------
.# 2026-07-09 13:34:49.48: <ERROR> External-process Error: mkmerge failed
External-process Error: see log /disk78/genOps/vsa/logs/TestVSAnjcVmcDeep/cu7id51.log
.# 2026-07-09 13:34:49.51: Updating archive curation history with this row:
.# 2026-07-09 13:34:49.51: (51, 7, '/disk78/genOps/vsa/logs/TestVSAnjcVmcDeep/cu7id51.log', 'NONE', '2026-07-09T12:34:49.51', 'njc', '[VMCDEEP] 4', 0)

**Comment by wfastrononomer** [comment]
Needed to define NoAperCorr columns.

New vmcdeepSource table complete.

Check it.

Need to commit changes to template schema, and to VSA_vmcdeepSchema.sql

**Comment by wfastrononomer** [comment]
Checks:

testVSAnjcVmcDeep> select COUNT(*) from vmcdeepSource
.------------------
.| |            | |
.------------------
.|>| 2444647    |<|
.------------------
Rows returned: 1 (execution time: 0.0116549 s)
testVSAnjcVmcDeep> select COUNT(*) from ramses27.VMCDEEPv20240506.dbo.vmcdeepSource
.------------------
.| |            | |
.------------------
.|>| 2444647    |<|
.------------------

Same number of sources

select COUNT(*) from ramses27.VMCDEEPv20240506.dbo.vmcdeepSource as so,vmcdeepSource as sn where so.sourceID=sn.sourceID
.------------------
.| |            | |
.------------------
.|>| 2444647    |<|
.------------------

Equal sourceIDs.
select COUNT(*) from ramses27.VMCDEEPv20240506.dbo.vmcdeepSource as so,vmcdeepSource as sn where so.sourceID=sn.sourceID and abs(sn.ksaperMag3-so.ksaperMag3)<0.00001
.------------------
.| |            | |
.------------------
.|>| 2444647    |<|
.------------------

select COUNT(*) from ramses27.VMCDEEPv20240506.dbo.vmcdeepSource as so,vmcdeepSource as sn where so.sourceID=sn.sourceID and abs(sn.ksaperMag3-so.ksaperMag3)<0.00001 and abs(sn.japerMag3-so.japerMag3)<0.00001
.------------------
.| |            | |
.------------------
.|>| 2444647    |<|
.------------------

All Ks aper mag3 match and all J aper mag 3 do so too.

**Comment by wfastrononomer** [comment]
select COUNT(*) from ramses27.VMCDEEPv20240506.dbo.vmcdeepSource as so,vmcdeepSource as sn where so.sourceID=sn.sourceID and abs(sn.ksaperMag2-so.ksaperMag3)<0.00001
.------------------
.| |            | |
.------------------
.|>| 263123     |<|
.------------------
Rows returned: 1 (execution time: 3.08625 s)
testVSAnjcVmcDeep> select COUNT(*) from ramses27.VMCDEEPv20240506.dbo.vmcdeepSource as so,vmcdeepSource as sn where so.sourceID=sn.sourceID and abs(sn.ksaperMag2-so.ksaperMag3)<0.00001 and sn.ksaperMag2>0.
.------------------
.| |            | |
.------------------
.|>| 445        |<|
.------------------

New aper mag2 column added and is different from aperMag3

select COUNT(*) from vmcdeepMergeLog as l,vmcdeepSource as s,vmcdeepDetectionPhotometry as p where l.frameSetID=s.frameSetID and p.multiframeID=l.ksmfID and p.extNum=l.kseNum and p.seqNum=s.ksseqNum and abs(p.aperMag3-s.ksaperMag3)>0.00001
.------------------
.| |            | |
.------------------
.|>| 0          |<|
.------------------
Rows returned: 1 (execution time: 0.942568 s)
testVSAnjcVmcDeep> select COUNT(*) from vmcdeepMergeLog as l,vmcdeepSource as s,vmcdeepDetectionPhotometry as p where l.frameSetID=s.frameSetID and p.multiframeID=l.ksmfID and p.extNum=l.kseNum and p.seqNum=s.ksseqNum and abs(p.aperMag2-s.ksaperMag3)>0.00001
.------------------
.| |            | |
.------------------
.|>| 2181524    |<|
.------------------
Rows returned: 1 (execution time: 0.821755 s)
testVSAnjcVmcDeep> select COUNT(*) from vmcdeepMergeLog as l,vmcdeepSource as s,vmcdeepDetectionPhotometry as p where l.frameSetID=s.frameSetID and p.multiframeID=l.ksmfID and p.extNum=l.kseNum and p.seqNum=s.ksseqNum and abs(p.aperMag2-s.ksaperMag2)>0.00001
.------------------
.| |            | |
.------------------
.|>| 0          |<|
.------------------

ksaperMag2 matches DetectionPhotometry values.

**Comment by wfastrononomer** [comment]
Need to check in schemas
r14120
and
r14121

However two issues. Some other columns missing - shouldn't have copied from vvv,vvvx
Also TilePawprints seem to have done something weird.

Look into these before copying over new vmcSource table.

**Comment by wfastrononomer** [comment]
TilePawprints are all from Deep tiles -> 11 tile sets only.
So they use the 6 offsets. but ProgrammeBuilder defines it from the stacks that typically have a offset 1-3.

Look at fix to sort this out.

**Comment by wfastrononomer** [comment]
nOffsets is incorrect for deep tiles - copied from inputs. 

Needs to be reset. 
Code worked when run on VSA, because there was no programmeID selection and it checked tiles across all programmes and took the max value.

But on test DB, only found these 4 deep tiles ( why 11 tile sets, not 4). 
One tileSet is default, 4 are good, 4 link to deprecated tile mIDs (but aren't deprecated)  and 2 don't have matches in release DB.

All have default tileSetIDs. 

VMCDEEPv20240506> select tileSetID,tlmfID,m.deprecated,ts.deprecated from vmcdeepTileSet as ts,Multiframe as m where ts.tlmfID=m.multiframeID
.---------------------------------------------------------------------------
.| | tileSetID           | tlmfID              | deprecated | deprecated | |
.---------------------------------------------------------------------------
.|>| -99999999           | -99999999           | 0          | 0          |<|
.|>| -99999996           | 1000000075253       | 104        | 0          |<|
.|>| -99999995           | 1000000075254       | 104        | 0          |<|
.|>| -99999994           | 1000000075351       | 104        | 0          |<|
.|>| -99999993           | 1000000075354       | 104        | 0          |<|
.|>| -99999992           | 1000000075355       | 0          | 0          |<|
.|>| -99999991           | 1000000075356       | 0          | 0          |<|
.|>| -99999990           | 1000000078164       | 0          | 0          |<|
.|>| -99999989           | 1000000078165       | 0          | 0          |<|
.---------------------------------------------------------------------------

**Comment by wfastrononomer** [comment]
Keep -99999999, -99999992, -99999991, -99999990, -99999989
Renumber -99999992 and larger.
Why values -ve?
Check code.

**Comment by wfastrononomer** [comment]
Need to:
delete from vmcdeepTileSet  where tileSetID>-99999999 and tileSetID<-99999992 but make VMCDEEPv20240506 read write first
Also apply to vmcdeepTilePawprints

**Comment by wfastrononomer** [comment]
VSA> select m.multiframeID,m.noffsets from ProgrammeFrame as p,Multiframe as m where p.programmeID=135 and p.productID>0 and p.releaseNum=4 and m.multiframeID=p.multiframeID and m.frameType like 'tiledeepstack%' and m.deprecated in (0,50)
.--------------------------------------
.| | multiframeID        | noffsets | |
.--------------------------------------
.|>| 1000000075355       | 3        |<|
.|>| 1000000075356       | 3        |<|
.|>| 1000000075357       | 3        |<|
.|>| 1000000075358       | 3        |<|
.|>| 1000000078163       | 3        |<|
.|>| 1000000078164       | 3        |<|
.|>| 1000000078165       | 3        |<|
.|>| 1000000078166       | 3        |<|
.--------------------------------------
Rows returned: 8 (execution time: 0.363133 s)
VSA> update Multiframe set noffsets=6 from ProgrammeFrame as p where p.programmeID=135 and p.productID>0 and p.releaseNum=4 and Multiframe.multiframeID=p.multiframeID and Multiframe.frameType like 'tiledeepstack%' and Multiframe.deprecated in (0,50)
Rows affected: 8 (execution time: 0.0112691 s)

**Comment by wfastrononomer** [comment]
Need to update in release.

**Comment by wfastrononomer** [comment]
Compared schemas with before my changes. Still not got vmcdeepSource correct.
Old one had Psf and See (all defaults - why) and was missing Petro and Kron which are in new.
r14126-14127 now have correct schema - diff between 13358 and 14126 shows this.

**Comment by wfastrononomer** [comment]
Created vmcdeepSource in testVSAnjcVmcDeep --> need to copy to release database.

**Comment by wfastrononomer** [comment]
And VSA.

**Comment by wfastrononomer** [comment]
Magnitudes look good.

Checked a few other values - look good.

**Comment by wfastrononomer** [comment]
VSA? Will foreign key constraints cause problems?

**Comment by wfastrononomer** [comment]
Added to release DB.
